packages/scheduler/src/__tests__/SchedulerPostTask-test.js

```diff
@@ -178,90 +189,115 @@ describe('SchedulerPostTask', () => {
   });
 
-  it('throws when a task errors then continues in a new event', () => {
+  it('an error in one task does not affect execution of other tasks', () => {
     scheduleCallback(NormalPriority, () => {
-      runtime.log('Oops!');
       throw Error('Oops!');
     });
     scheduleCallback(NormalPriority, () => {
       runtime.log('Yay');
     });
-    runtime.assertLog(['Post Task']);
-
-    expect(() => runtime.fireNextTask()).toThrow('Oops!');
-    runtime.assertLog(['Task Event', 'Oops!', 'Post Task']);
-
-    runtime.fireNextTask();
-    runtime.assertLog(['Task Event', 'Yay']);
+    runtime.assertLog([
+      'Post Task 0 [user-visible]',
+      'Post Task 1 [user-visible]',
+    ]);
+    runtime.flushTasks();
+    runtime.assertLog([
+      'Task 0 Fired',
+      'Task 0 errored [Oops!]',
+      'Task 1 Fired',
+      'Yay',
+    ]);
   });
 
   it('schedule new task after queue has emptied', () => {
     scheduleCallback(NormalPriority, () => {
       runtime.log('A');
     });
 
-    runtime.assertLog(['Post Task']);
-    runtime.fireNextTask();
-    runtime.assertLog(['Task Event', 'A']);
+    runtime.assertLog(['Post Task 0 [user-visible]']);
+    runtime.flushTasks();
+    runtime.assertLog(['Task 0 Fired', 'A']);
 
     scheduleCallback(NormalPriority, () => {
       runtime.log('B');
     });
-    runtime.assertLog(['Post Task']);
-    runtime.fireNextTask();
-    runtime.assertLog(['Task Event', 'B']);
+    runtime.assertLog(['Post Task 1 [user-visible]']);
+    runtime.flushTasks();
+    runtime.assertLog(['Task 1 Fired', 'B']);
   });
 
   it('schedule new task after a cancellation', () => {
     const handle = scheduleCallback(NormalPriority, () => {
       runtime.log('A');
     });
 
-    runtime.assertLog(['Post Task']);
+    runtime.assertLog(['Post Task 0 [user-visible]']);
     cancelCallback(handle);
 
-    runtime.fireNextTask();
-    runtime.assertLog(['Task Event']);
+    runtime.flushTasks();
+    runtime.assertLog([]);
 
     scheduleCallback(NormalPriority, () => {
       runtime.log('B');
     });
-    runtime.assertLog(['Post Task']);
-    runtime.fireNextTask();
-    runtime.assertLog(['Task Event', 'B']);
+    runtime.assertLog(['Post Task 1 [user-visible]']);
+    runtime.flushTasks();
+    runtime.assertLog(['Task 1 Fired', 'B']);
+  });
+
+  it('schedules tasks at different priorities', () => {
+    scheduleCallback(ImmediatePriority, () => {
+      runtime.log('A');
+    });
+    scheduleCallback(UserBlockingPriority, () => {
+      runtime.log('B');
+    });
+    scheduleCallback(NormalPriority, () => {
+      runtime.log('C');
+    });
+    scheduleCallback(LowPriority, () => {
+      runtime.log('D');
+    });
+    scheduleCallback(IdlePriority, () => {
+      runtime.log('E');
+    });
+    runtime.assertLog([
+      'Post Task 0 [user-blocking]',
+      'Post Task 1 [user-blocking]',
+      'Post Task 2 [user-visible]',
+      'Post Task 3 [user-visible]',
+      'Post Task 4 [background]',
```

Comment:
Is it worth testing the opposite order? I guess that would only be testing that our mock handled the ordering correctly, but that may give us more confidence in the mock.

Reply:
The mock intentionally doesn’t sort the tasks, just flushes the whole queue. The spec doesn’t make any semantic guarantees on ordering. I consider these browser integration tests, not tests of `postTask`.